cms_coding_rules.md

```diff
@@ -0,0 +1,144 @@
+# CMS Naming, Coding, And Style Rules
+
+### By the CMS Ofﬂine Software Development Team
+
+## Abstract
+This document describes the design, naming, coding, and style rules and recommendations for CMS software written in C++, plus a few guidelines for Python configuration.
+
+
+## Outline
```

Comment:
This could be a real table of contents with links to the anchors for each section, e.g.:
```Markdown
[1 -- Introduction](#1----introduction)
```

Reply:
I added these links. The syntax is actually:
[1 -- Introduction](#1----introduction-1)